src/main/java/Control/Scenario/UC_NewProject.java

```diff
@@ -0,0 +1,173 @@
+            jsonResponse.setMessage("Internal server error.");
+        } catch (AlreadyExistsException e) {
+            db.afterExceptionInSqlExecution(e);
+
+            jsonResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            jsonResponse.setMessage("Project with this name already exists.");
+        }
+
+        return jsonResponse;
+    }
+
+    // Privates
+    private void insertProjectAndProjectUsersIntoDatabase(@NotNull final ProjectCreation projectCreation, @NotNull final List<T_User> listOfUsers) throws SQLException{
+        // Insert new Project
+        I_Project.insert(db.getConn(), db.getPs(), T_Project.CreateFromScratch(projectCreation.getProject_name()));
+
+        int latestId = I_Project.retrieveLatestPerConnectionInsertedID(db.getConn(), db.getPs(), db.getRs());
+
+        // Insert new Connections between project and user
+        for (T_User user:
+             listOfUsers) {
+            Dictionary dict = new Hashtable();
+
+            dict.put(T_Project_user.DBNAME_PROJECTID, latestId);
+            dict.put(T_Project_user.DBNAME_USERID, user.getA_pk());
+
+            I_ProjectUser.insert(db.getConn(), db.getPs(), T_Project_user.CreateFromScratch(dict));
+        }
+    }
+
+    private List<T_User> getAllUsersThatWillBeOwners(@NotNull final ProjectCreation projectCreation) throws SQLException{
+        List<T_User> list = new ArrayList<>();
+
+        // process required email
+        T_User requiredUser = I_User.retrieveByEmail(db.getConn(), db.getPs(), db.getRs(), projectCreation.getRequired_email());
+
+        if (requiredUser == null)
+            return null;
+
+        list.add(requiredUser);
+
+        // process emails of additional users
+        for (String email:
+             projectCreation.getAdditional_emails()) {
+            T_User tempUser = I_User.retrieveByEmail(db.getConn(), db.getPs(), db.getRs(), email);
+
+            // never allow null (all emails inserted have to be present)
+            if (tempUser == null)
+                return null;
+
+            list.add(tempUser);
+        }
+
+        return list;
+    }
+
+
+    /***
+     * Project name has to be valid, emails have to be valid: basic requirements
+     * @param projectCreation
+     * @return
+     */
+    private boolean isProjectNameAndEmailsValid(@NotNull final ProjectCreation projectCreation) {
+
+        // Valid project name
+        if (projectCreation == null || projectCreation.getProject_name().equals(""))
+            return false;
+
+        // All email's are valid
+        String required = projectCreation.getRequired_email();
+
+        boolean requiredPresent = EmailValidator.getInstance().isValid(required);
+
+        for (String em:
+                projectCreation.getAdditional_emails()) {
+            if (EmailValidator.getInstance().isValid(em) == false)
+                return false;
+        }
+
+        return requiredPresent;
```

Comment:
nemusi sa to menit len nerozumiem nazvu requiredPresent, naznacuje to ze existuje pozadovany parameter?

Reply:
To je len efektívne nakódené, že je potrebné aby ten required bol validny a pokiaľ nejaky z pridavnych neni validny tak to hodime rovno false vieš